File: src/main/java/de/tum/cit/aet/artemis/core/dto/CoursesForDashboardDTO.java
package de.tum.cit.aet.artemis.core.dto;

import java.util.Set;

import com.fasterxml.jackson.annotation.JsonInclude;

import de.tum.cit.aet.artemis.exam.domain.Exam;

@JsonInclude(JsonInclude.Include.NON_EMPTY)
public record CoursesForDashboardDTO(Set<CourseForDashboardDTO> courses, Set<Exam> activeExams) {

}

File: src/main/webapp/app/core/course/manage/courses-for-dashboard-dto.ts
import { CourseForDashboardDTO } from 'app/core/course/manage/course-for-dashboard-dto';
import { Exam } from 'app/entities/exam/exam.model';

export class CoursesForDashboardDTO {
    courses: CourseForDashboardDTO[];
    activeExams: Exam[];
}
